response-types.js

```diff
@@ -1,27 +1,79 @@
-class Response {
-  constructor (data, meta = undefined) {
-    this.data = data
-    this.errors = []
-    if (meta) { this.meta = meta }
-  }
-}
+const statuses = require('statuses')
+const util = require('./utils/util')
+
+const codes = [200, 201, 202]
 
-class Error {
-  constructor (title) {
+class FSError extends Error {
+  constructor ({code, title, detail = '', status = '500'}, extraProps = {}) {
+    super(detail === '' ? title : detail)
+    this.code = code
     this.title = title
+    this.status = status
+    this.detail = detail
+    Object.keys(extraProps).forEach(key => {
+      this[key] = extraProps[key]
+    })
+  }
+
+  static unknown () {
+    return new FSError({code: 'FS-ERR-0', title: 'An error occurred'})
+  }
+
+  static userDoesNotExist ({detail = '', status = '404'} = {}) {
+    return new FSError({code: 'FS-ERR-1', title: 'User doesn\'t exist', detail, status})
+  }
+
+  static missingParameters ({errors, status = '400'}) {
+    errors[0] = util.capitalize(errors[0])
+    return new FSError({code: 'FS-ERR-2', title: 'Missing parameters', detail: `${errors.join(', ')} is invalid`, status})
+  }
+
+  static fieldAlreadyExists ({detail, status = '409'}) {
+    return new FSError({code: 'FS-ERR-3', title: 'Field already exists', detail, status})
+  }
+
+  static unauthorized ({detail = '', status = '401'} = {}) {
+    return new FSError({code: 'FS-ERR-4', title: 'Bad token', detail, status})
+  }
+
+  static invalidPassword ({status = '403'} = {}) {
+    return new FSError({code: 'FS-ERR-5', title: 'Invalid password', status})
+  }
+
+  static userIsNotLoggedIn ({status = '412'} = {}) {
+    return new FSError({code: 'FS-ERR-6', title: 'User is not logged in', status})
   }
 }
 
-class ErrorResponse extends Response {
-  /**
-   * Generates error response
-   * @param {Error[]} errors
-   * @param {Object?} meta
-   */
-  constructor (errors, meta = undefined) {
-    super(null, meta)
-    this.errors = errors
+// Only created like this for code completion
+let Response = {
+  OK: sendFakeResponse,
+  Created: sendFakeResponse,
+  Accepted: sendFakeResponse
+}
+Response = statuses.codes.filter(code => codes.some(usedCode => code === usedCode)).map(code => {
+  // Generates each function
+  const functionName = statuses[code].replace(/[^a-zA-Z]/g, '')
+  if (code >= 400 && code < 600) {
```

Comment:
Like we talked about, this can be removed 👍🏽

Reply:
Yep, the error functions will be removed